Document/0x04d-Security-Testing-SDLC.md

```diff
@@ -38,29 +38,44 @@ From the human aspect, this is achieved by creating cross functional teams that
+- Then, as architecture and design are ongoing, a foundational artefact must be performed: **Threat Modeling**, which is b asically an activity where threats are identified, enumerated, prioritized and their treatment initialized. An input of the Threat Model is the **Security Architecture**, but which can be refined after Threat Modeling (both for software and hardware aspects). **Secure Coding rules** are established and the list of **Security tools** that will be used is created. Also, the strategy for **Security testing** is clarified;
+- All security requirements and design considerations should be stored in the Application Life cycle Management System (ALM), which is typically known as issue tracker, that the development/ops team already uses to ensure that security requirements are tightly integrated into the development workflow. The security requirements should ideally also contain the relevant source code snippets for the used programming language, to ensure that developers can quickly reference them. Another strategy for secure coding guidelines is to create a dedicated repository under versioning control, that only contains these code snippets, which has many benefits over the traditional approach of storing these guidelines in word documents or PDFs.
+- The next step is to **develop software**, including security activities such as **Code Reviews** (usually with peers), **Static Analysis** with automated tools and **Unit Tests** dedicated to security;
 - Then comes the long-awaited moment to perform tests on the release candidate: **Penetration Testing** ("Pentests"), using both manual and automated techniques;
 - And finally, after software has been **Accredited** by all stakeholders, it can be transitioned to Operation teams and safely put in Production.
 
 The picture below shows all the phases with the different artefacts:
 -- TODO [Add a picture of a SDLC diagram that clarifies the description above] --
 
-Based on the risks of the project, some artefacts may be simplified (or even skipped) while others may be added (formal intermediary approvals, documentation of certain points, ...). **Always keep in mind a SDLC is meant to bring risk reduction to software development and is a framework that helps put in place controls that will reduce those risks to an acceptable level. ** While this is a generic description of SDLC, always tailor this framework to the needs of your projects.
+Based on the risks of the project, some artefacts may be simplified (or even skipped) while others may be added (formal intermediary approvals, formal documentation of certain points, ...). **Always keep in mind a SDLC is meant to bring risk reduction to software development and is a framework that helps put in place controls that will reduce those risks to an acceptable level. ** While this is a generic description of SDLC, always tailor this framework to the needs of your projects.
```

Comment:
Can you please fix the formatting? The rest of the chapter is bold. The "**" should come directly after "level" or "projects" without a space.